File: Facade/c#/Facade.cs
class ProviderCommunication
{
    public void Receive() => Console.WriteLine("receiving products from the manufacturer");

    public void Payment() => Console.WriteLine("payment");
}

class Site
{
    public void Placement() => Console.WriteLine("placement on the website");

    public void Del() => Console.WriteLine("removal from the site");
}

class Database
{
    public void Insert() => Console.WriteLine("writing to the database");

    public void Del() => Console.WriteLine("deleting from the database");
}

class MarketPlace
{
    private ProviderCommunication _providerCommunication;

    private Site _site;
    
    private Database _database;

    public MarketPlace()
    {
        _providerCommunication = new ProviderCommunication();
        _site = new Site();
        _database = new Database();
    }

    public void ProductReceipt()
    {
        _providerCommunication.Receive();
        _site.Placement();
        _database.Insert();
    }

    public void ProductRelease()
    {
        _providerCommunication.Payment();
        _site.Del();
        _database.Del();
    }
}

class Program
{
    static void Main()
    {
        MarketPlace marketPlace = new MarketPlace();
        
        marketPlace.ProductReceipt();
        
        Console.WriteLine();
        
        marketPlace.ProductRelease();
    }
}

File: Facade/python/Facade.py
class ProviderCommunication:

    def receive(self):
        print('receiving products from the manufacturer')

    def payment(self):
        print('payment')


class Site:

    def placement(self):
        print('placement on the website')

    def delete(self):
        print('removal from the site')


class Database:

    def insert(self):
        print('writing to the database')

    def delete(self):
        print('deleting from the database')


class MarketPlace:

    def __init__(self):
        self._provider_communication = ProviderCommunication()
        self._site = Site()
        self._database = Database()

    def product_receipt(self):
        self._provider_communication.receive()
        self._site.placement()
        self._database.insert()

    def product_release(self):
        self._provider_communication.payment()
        self._site.delete()
        self._database.delete()


if __name__ == '__main__':

    market_place = MarketPlace()
    market_place.product_receipt()
    print()
    market_place.product_release()